### PR TITLE
ref(similarity-embeddings): Modify existing similarity UI

### LINKED ISSUE #62240: Change existing similarity UI to use numbers
Change the existing similarity UI in the following ways:

- Convert Exception & Message UI to numbers (i.e. remove the bars UI)
- Remove color legend
- Remove the hover experience on “Exception” and “Message”
- Remove the button to show issues below the threshold, we’ll just show everything

### PR DESCRIPTION
**Pull request by jangjodi**
Change the existing similarity UI in the following ways:
- Convert Exception & Message UI to numbers (i.e. remove the bars UI)
- Remove color legend
- Remove the hover experience on “Exception” and “Message”
- Remove the button to show issues below the threshold, we’ll just show everything

with flag:
Screenshot 2023-12-27 at 9 25 04 AM

without flag:
Screenshot 2023-12-27 at 11 53 23 AM

closes https://github.com/getsentry/sentry/issues/62240

### CODE REVIEW

File: static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/index.tsx [hunk 1]
```diff
@@ -47,6 +47,9 @@ function SimilarStackTrace({params, location, project}: Props) {
   const navigate = useNavigate();
   const prevLocationSearch = usePrevious(location.search);
   const hasSimilarityFeature = project.features.includes('similarity-view');
+  const hasSimilarityEmbeddingsFeature = project.organization?.features?.includes(
```
Comment (jangjodi) [inline static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/index.tsx:50]: is there another way to get organization if the `project.organization` does not exist?

File: static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/index.tsx [hunk 2]
```diff
@@ -158,13 +163,26 @@ function SimilarStackTrace({params, location, project}: Props) {
...
             project={project}
+            organization={organization}
+            groupId={groupId}
+            pageLinks={items.pageLinks}
+          />
+        )}
+        {status === 'ready' && hasSimilarItems && hasSimilarityEmbeddingsFeature && (
+          <List
+            items={items.similar.concat(items.filtered)}
+            filteredItems={[]}
```
Comment (jangjodi) [inline static/app/views/issueDetails/groupSimilarIssues/similarStackTrace/index.tsx:181]: combine similar items and filtered items into `items` to show everything regardless of if it's below the threshold

File: static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx [hunk 3]
```diff
@@ -47,18 +51,36 @@ describe('Issues Similar View', function () {
...
     });
+    mockOrganization();
   });
 
   afterEach(() => {
     MockApiClient.clearMockResponses();
     jest.clearAllMocks();
   });
 
+  const selectNthSimilarItem = async index => {
```
Comment (scttcper) [inline static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx:72]:
```suggestion
  const selectNthSimilarItem = async (index: number) => {
```

File: static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx [hunk 4]
```diff
@@ -47,18 +51,36 @@ describe('Issues Similar View', function () {
...
+    const items = await screen.findAllByTestId('similar-item-row');
+
+    const item = items.at(index);
+
+    if (item === undefined) {
+      throw new Error('Invalid item index');
+    }
```
Comment (scttcper) [inline static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx:79]:
```suggestion
    expect(item).toBeDefined()
```

**Review by scttcper — APPROVED** [review]

File: static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx [hunk 5]
```diff
@@ -47,18 +51,34 @@ describe('Issues Similar View', function () {
...
   });
 
+  const selectNthSimilarItem = async (index: number) => {
+    const items = await screen.findAllByTestId('similar-item-row');
+
+    const item = items.at(index);
+
+    expect(item).toBeDefined();
+
+    await userEvent.click(item);
```
Comment (scttcper) [inline static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx:79]:
```suggestion
    await userEvent.click(item!);
```

File: static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx [hunk 6]
```diff
@@ -47,18 +51,36 @@ describe('Issues Similar View', function () {
 
   const router = RouterFixture();
 
+  function mockOrganization(props?: {features: string[]}) {
```
[the remark this answers is not in the document]
Comment (jangjodi) [reply static/app/views/issueDetails/groupSimilarIssues/index.spec.tsx]: thanks! I didn't know about this

**Review by malwilley — APPROVED** [review]

**Pull request merged**